include/verilated_random.h

```diff
@@ -38,15 +38,18 @@ class VlRandomVar final : public VlRandomExpr {
     const char* const m_name;  // Variable name
     void* const m_datap;  // Reference to variable data
     const int m_width;  // Variable width in bits
+    const unsigned m_randModeIdx;  // rand_mode index
 
 public:
-    VlRandomVar(const char* name, int width, void* datap)
+    VlRandomVar(const char* name, int width, void* datap, unsigned randModeIdx)
         : m_name{name}
         , m_datap{datap}
-        , m_width{width} {}
+        , m_width{width}
+        , m_randModeIdx{randModeIdx} {}
     const char* name() const { return m_name; }
     int width() const { return m_width; }
     void* datap() const { return m_datap; }
+    unsigned randModeIdx() const { return m_randModeIdx; }
```

Comment:
```suggestion
    unsigned randModeIdx() const { return m_randModeIdx; }
    bool randModeIdxNone() const { return m_randModeIdx == std::numeric_limits<unsigned>::max(); }
```

To abstract away the logic above that looks for the magic value.